Dev

Arvi3d opened the pull request.

ArtsiomCh committed:
Experimental login, load file and check analysis result.

ArtsiomCh committed:
Persistent login data, Notifications unification, File extension compatibility check.

ArtsiomCh committed:
Panel with simple editor(view only) for analysis results added.

ArtsiomCh committed:
Centralised `Get Analysis` invocation and cached results; Panel's editor updated.

ArtsiomCh committed:
experimental externalAnnotator for suggestions highlighting.

ArtsiomCh committed:
local cache enabled and cleanup implementation; repetitive checking for analysis results done.

ArtsiomCh committed:
Presentable format for messages at Panel; some refactoring.

ArtsiomCh committed:
messages at Panel sorted by source file appearance order.

ArtsiomCh committed:
Panel updates when new file chosen in the editor.

ArtsiomCh committed:
refactoring and some additions for future `Project Files` suggestions Panel

ArtsiomCh committed:
refactoring and milty-file upload/retrieve results implementation.

ArtsiomCh committed:
race condition for Cache write/remove fixed; UI thread freezes by Annotator fixed.

ArtsiomCh committed:
fix UI thread freezes when updating CurrentFile Panel.

ArtsiomCh committed:
draft of new Panel made from ToDo plugin sources.

ArtsiomCh committed:
Analyse all project files action implementation.

ArtsiomCh committed:
DeepCode.ai group in ToolsMenu

ArtsiomCh committed:
break in few createBundle requests if size of analysing files > MAX_BUNDLE_SIZE

ArtsiomCh committed:
fallback to .create****Annotation methods for pre 2020 versions compatibility

ArtsiomCh committed:
fix to show suggestions in the panel for current Project only

ArtsiomCh committed:
do update Panel if file was edited

ArtsiomCh committed:
do show expand/collapse action on Panel's toolbar

ArtsiomCh committed:
fix Group By actions group on Panel's toolbar

ArtsiomCh committed:
do show Suggestion text with corresponding links to the source as children in Panel

ArtsiomCh committed:
do show severity icon for suggestions in Panel

ArtsiomCh committed:
QuickFix to Suppress current suggestion in PopUp

ArtsiomCh committed:
QuickFix to Suppress suggestion in current file

ArtsiomCh committed:
do suppression(ignoring) for two or more suggestions at the same spot.

ArtsiomCh committed:
do place cursor and select ignoring description template.

ArtsiomCh committed:
fix suggestions tree rebuild for file lost all source links except last one.

ArtsiomCh committed:
add Analyse Project button to Panel's toolbar.

ArtsiomCh committed:
default Flat View for the Panel's Tree and expand to files level.

ArtsiomCh committed:
fix UI freeze while Analyse Project action executing.

ArtsiomCh committed:
new action to Invalidate cache (for current project) and re-analyse project files.

ArtsiomCh committed:
remove unneeded actions from Toolbar and Invalidate Cache action optimisation

ArtsiomCh committed:
presentation of errors, warnings and infos at Summary Node.

ArtsiomCh committed:
generalise async call for Panel update

ArtsiomCh committed:
colored presentation of errors, warnings and infos at Summary Node and File nodes.

ArtsiomCh committed:
sorting by severities/number of issues for FileNodes and by severity for SuggestionNodes

ArtsiomCh committed:
Status bar current status and Progress indicator shown for Analyse Project action.

ArtsiomCh committed:
do show summary Icons set with numbers at Panel Tab.

ArtsiomCh committed:
icons scaling and (potential) numbers colorization

ArtsiomCh committed:
logging process adopted and generalized for baseUrl changing.

ArtsiomCh committed:
Settings Panel with baseUrl changing functionality

ArtsiomCh committed:
`severity`, `linters`, `session token` params fully implemented; Login process and notifications revised; show Settings action

ArtsiomCh committed:
session token validity checks, notifications fixes, trailing `/` for baseUrl check

ArtsiomCh committed:
fileHash implementation of CreateBundle and UploadFiles usage

ArtsiomCh committed:
ExtendBundle implemented: 1 project = 1 actual bundle; no restrictions for {filePath:fileHash} size (4Mb chunks rule)

ArtsiomCh committed:
Make runnable on Idea, WebStorm, PyCharm 2019.2 and up.

ArtsiomCh committed:
console output logging format improving.

ArtsiomCh committed:
some more console output logging format improving.

ArtsiomCh committed:
configFiles contains filename check added

ArtsiomCh committed:
ignore files over 5MB

ArtsiomCh committed:
show file's Path in Panel

ArtsiomCh committed:
analyse on startup enabled and interface cleanup from unneeded(debug) buttons and popups

ArtsiomCh committed:
fix suppressing availability in preview editor (it will close, as no suggestion found, before description entered).

ArtsiomCh committed:
Panel's summary Icon set with `?` while analysis is in progress.

ArtsiomCh committed:
fix ToolWindow summary icon update if ToolWindow is not open.

ArtsiomCh committed:
user and dev guides added; logo icon added; minor improvements.

ArtsiomCh committed:
language="textmate" support added (`.ts` at PyCharm); dev guide additions.

ArtsiomCh committed:
Open analysis results in browser action added

ArtsiomCh committed:
remove file from bundle when file was removed from project. !Need correction! (see added todo)

ArtsiomCh committed:
make "Show results in browser" enable/disable depending on results availability.

ArtsiomCh committed:
Login: new token requested after user click the link; optimisation of flow and parameters setup.

ArtsiomCh committed:
isLogged checking in a loop after user sent to the browser.

ArtsiomCh committed:
caches cleanup implementation through projectListener.

ArtsiomCh committed:
full proceeding of files created, deleted and changed outside of IDE; file deletion reflected on server too.

ArtsiomCh committed:
fix listeners to work at 192 (move from plugin.xml to StartupActivity)

ArtsiomCh committed:
informative progress indicator and fixed most of freezes.

ArtsiomCh committed:
fix filePath MacOS/Linux compatibility and it Panel's presentation

ArtsiomCh committed:
.gitignore files/dirs excluded from analysis

Arvi3d commented:
retry deepcode

Arvi3d commented:
retry deepcode

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/DeepCodeAI/jetbrains-plugin/f05afa6ac3498ada8e39d45d25173e226eca83c8/DeepCodeAI/jetbrains-plugin/c9ca90e56bf9acae4914419d2faa7fe34e9ae0b6/pr/_/%2F/code/?utm_source=gh_review?c=0&w=0&i=0) your code in 0.190 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

##### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/DeepCodeAI/jetbrains-plugin/f05afa6ac3498ada8e39d45d25173e226eca83c8/DeepCodeAI/jetbrains-plugin/c9ca90e56bf9acae4914419d2faa7fe34e9ae0b6/pr/_/%2F/code/?utm_source=gh_review?c=0&w=0&i=0)
---

ArtsiomCh committed:
MyBulkFileListener refactoring

ArtsiomCh committed:
newLogin request with user-agent as IdeName-Version

ArtsiomCh committed:
fix reusing token when settings copied from one Ide to another.

ArtsiomCh commented:
@Arvi3d have a look please: all links at the dclocal bot's message above lead to `localhost:8080`.

Arvi3d commented:
Все все правильно. То специальный бот для локальной отладки

> On 1 May 2020, at 12:46, Artsiom Chapialiou <notifications@github.com> wrote:
> 
> 
> @Arvi3d <https://github.com/Arvi3d> have a look please: all links at the dclocal bot's message above lead to localhost:8080.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/DeepCodeAI/jetbrains-plugin/pull/1#issuecomment-622340239>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAA5QVONWK3FESX73XLZSKTRPKR7NANCNFSM4MVIBLMA>.
>

ArtsiomCh committed:
fix login Notifications: show application wide (re-login requests on Url/token change), show on login success, disable expired old requests.

ArtsiomCh committed:
Consent to analyse request implemented; notifications refactored.

ArtsiomCh committed:
fix compatibility with 2020.1.1 (https://youtrack.jetbrains.com/issue/IDEA-239960); refactoring of core AnalysisData

ArtsiomCh committed:
.dcignore basic glob syntax: file and dir/ names, *

ArtsiomCh committed:
rescan on .gitignore changes; fix .gitignore content delay to be parsed https://youtrack.jetbrains.com/issue/IDEA-239773

ArtsiomCh committed:
set focus on editor if called from DeepCode Panel

ArtsiomCh committed:
fix consent notification presentation

ArtsiomCh committed:
fix presentation for projects with no supported files; fix result's web-link shown for previous project

ArtsiomCh committed:
fix deadlocks: Mutex to update main cache, ReadAction for BulkFileListener; logging refactoring;

ArtsiomCh committed:
fix deadlocks: added checkCanceled in parsing results; remove unneeded ReadAction for BulkFileListener; threads invocation logging;

ArtsiomCh committed:
fix deadlocks: SmartMode (DumbAware) for all ReadActions to avoid UI freeze on indexes update.

ArtsiomCh committed:
.dcignore syntax implementation by https://git-scm.com/docs/gitignore#_pattern_format

ArtsiomCh committed:
inserting suppress comment: use the same offset as commented line.

ArtsiomCh committed:
inserting suppress comment: use language specific line comment prefix.

ArtsiomCh committed:
logging improvements: warn level added; more logging events for empty project's files list.

ArtsiomCh committed:
remove redundant ` ` for python comment

ArtsiomCh committed:
move from ExternalAnnotator to Annotator; RunUtils for backgrounds/RW access/smart mode

ArtsiomCh committed:
smart/dumb mode logging

ArtsiomCh committed:
split Update and Get operations for cached Analysis results;
back to ExternalAnnotator (to work on files level);
fix working with (and log it) ProgressIndicators;

ArtsiomCh committed:
fix race conditions for simultaneous file change (user typing);
stop invoked Background processes on project closing while analysis update is in progress;

ArtsiomCh committed:
prevent cancelling already stopped task/process

ArtsiomCh committed:
highlighting for all supported extensions (hardcoded Language's refs)

ArtsiomCh committed:
fix status bar info hiding for startup analysis.

ArtsiomCh committed:
add more logging for Login process to catch flicky bags;
refactoring - split all Login methods into separate class.

ArtsiomCh committed:
fix NullPointerException when checking ProgressIndicator with no text

ArtsiomCh committed:
fix `message doubled` for .js files

ArtsiomCh committed:
stop background jobs before clearing cache (fix deadlock if url changed while file uploading)

ArtsiomCh committed:
fix deepcode.ai url

Commit:
git ignored .gradle folder

ArtsiomCh committed:
fix WS did not cancel prev indicator;
fix missing indicator cancellations from runInBackgroundCancellable but before runInBackground.

Commit:
updated README file, merged with userGuide;
added label beta to plugin's name;
git ignored build folder;
updated plugin's description

Commit:
Merge branch 'dev' of github.com:DeepCodeAI/jetbrains-plugin into dev

ArtsiomCh committed:
fix deadlocks due to too many consequent requests for bulk Psi updates.

ArtsiomCh committed:
refactoring by deepcode.ai suggestions.

ArtsiomCh commented:
retry deepcode

Commit:
bumped version, reseted changelog
